src/dc_msg.rs

```diff
@@ -1546,4 +1563,36 @@ mod tests {
             free(mime_0 as *mut libc::c_void);
         }
     }
+
+    #[test]
+    pub fn test_prepare_message_and_send() {
+        use crate::config::Config;
+
+        unsafe {
+            let d = test::dummy_context();
+            let ctx = &d.ctx;
+
+            let contact = dc_create_contact(
+                ctx,
+                b"\x00".as_ptr().cast(),
+                b"dest@example.com\x00".as_ptr().cast(),
```

Comment:
You could also write this as `"dest@example.com".strdup()` I think (and above `"".strdup()`).

But TIL about `.cast()` though, nice.

Reply:
`strdup` performs allocation, and must be `free`d manually. I'd rather have Rust do it for me.

Reply:
PS. Since there is goal to drop unstable feature, `cast` will be gone eventually. Sad, I liked him.

Reply:
@KAction  `cast` is stable now: https://github.com/rust-lang/rust/pull/62713